linked_list.rb

```diff
@@ -42,17 +42,41 @@ def search(value)
   # method to return the max value in the linked list
   # returns the data value and not the node
   def find_max
-    puts "Not implemented"
+    current = @head
+    max = current.data
+    while current.next
+      current = current.next
+      if current.data > max
+        max = current.data
+      end
+    end
+    return max
   end
 
   # method to return the min value in the linked list
   # returns the data value and not the node
   def find_min
+    current = @head
+    min = current.data
+    while current.next
+      current = current.next
+      if current.data < min
+        min = current.data
+      end
+    end
+    return min
     puts "Not implemented"
   end
 
   # method that returns the length of the singly linked list
   def length
+    counter = 1
+    current = @head
+    while current.next

```

Comment:
Always check for null. :-) What if the linked list is empty? In that case, the length returned should be 0.